### PR TITLE
add obfbytes support

### PR DESCRIPTION
**Pull request by Yneth**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -238,33 +238,81 @@ macro_rules! obfstr {
...
+		buf
+	}};
+	($s:expr) => {{
+		const _OBFSTR_BYTES: &[u8] = $s;
+		const _OBFSTR_LEN: usize = _OBFSTR_BYTES.len();
+		const _OBFSTR_KEYSTREAM: [u8; _OBFSTR_LEN] = $crate::bytes::keystream::<_OBFSTR_LEN>($crate::random!(u32));
+		static _OBFSTR_DATA: [u8; _OBFSTR_LEN] = $crate::bytes::obfuscate::<_OBFSTR_LEN>(_OBFSTR_BYTES, &_OBFSTR_KEYSTREAM);
+		&$crate::bytes::deobfuscate::<_OBFSTR_LEN>($crate::xref(&_OBFSTR_DATA, $crate::random!(usize) & 0xffff), &_OBFSTR_KEYSTREAM)
+	}};
+    ($(let $name:ident = $s:expr;)*) => {$(
```
Comment (Yneth) [inline src/lib.rs:259]: I am using spaces instead of tabs

File: src/lib.rs [hunk 2]
```diff
@@ -238,33 +238,81 @@ macro_rules! obfstr {
...
+    #[test]
+    fn test_obfbytes_const() {
+        assert_eq!(obfbytes!(b"\x01\xa9"), b"\x01\xa9");
+    }
 
-	assert_eq!(obfstr!(L ABC), &[b'A' as u16, b'B' as u16, b'C' as u16]);
-	assert_eq!(obfstr!(L WORLD), &[0xd83c, 0xdf0d]);
+    #[test]
+    fn test_obfstr_let() {
+        obfstr! {
```
Comment (Yneth) [inline src/lib.rs:289]: it would be better for you to reformat, did not want to make PR huge

**Comment by Yneth** [comment]
Already saw someone requested this feature, and I need it too

**Comment by CasualX** [comment]
Hey thanks for the PR!

Can you show a concrete case for which you'd use obfbytes? I was thinking about an issue where the result of obfbytes is a potentially unaligned deobfuscated byte array where you'd want to reinterpret it to some larger structure.

I'm currently busy with other things so it may take some time for me to get around to implement this.

**Comment by Yneth** [comment]
I need to obfuscate byte array which cannot be represented as a valid UTF-8 string.

Actually, I found a way how to overcome my problem by the use of base64, such that I obfuscate the encoded string, 
so I am not sure if this PR is really that useful.

**Pull request closed without merging**